src/components/PostDetails.tsx

```diff
@@ -1,118 +1,69 @@
+/* eslint-disable import/extensions */
 import React, { useEffect, useState } from 'react';
+import {
+  getCommentPosts,
+  addComment,
+  deleteCom,
+} from '../features/CommentSlice';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { CommentData } from '../types/Comment';
 
-import * as commentsApi from '../api/comments';
+export const PostDetails: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { selectedPost } = useAppSelector(state => state.selectedPost);
+  const post = selectedPost;
+  const { comments, loading, error } = useAppSelector(state => state.comments);
 
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
-
-type Props = {
-  post: Post;
-};
-
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
   const [visible, setVisible] = useState(false);
 
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi
-      .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
+  useEffect(() => {
     setVisible(false);
-    setError(false);
 
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
+    if (post) {
+      dispatch(getCommentPosts(post.id));
     }
-  };
+  }, [post?.id, dispatch, post]);
```

Comment:
post is already in dependecies
```suggestion
  }, [dispatch, post]);
```